sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -239,10 +248,11 @@
 
     @AutoValue.Builder
     abstract static class Builder {
-      abstract Builder setProjectId(String projectId);
-      abstract Builder setQuery(Query query);
-      abstract Builder setNamespace(String namespace);
-      abstract Builder setNumQuerySplits(int numQuerySplits);
+      abstract Builder setProjectId(ValueProvider<String> projectId);
+      abstract Builder setQuery(ValueProvider<Query> query);
```

Comment:
How can the Query object be specified in the template? Can it be read from a String? If not, why does this need to be a value provider?